backend/app2/services/user_service.py

```
@@ -140,3 +142,78 @@ async def change_password(
+            ValueError: If the file type is not supported.
+        """
+        logger.info(f"Updating avatar for user ID: {user_id}")
+
+        # Validate file type
+        allowed_content_types = ["image/jpeg", "image/png", "image/gif"]
+        if file.content_type not in allowed_content_types:
+            logger.error(f"Invalid file type for avatar: {file.content_type}. User ID: {user_id}")
+            raise ValueError(f"Invalid file type: {file.content_type}. Please upload a JPG, PNG, or GIF.")
+
+        # Ensure user profile exists (also fetches it for returning later)
+        profile = await self.get_profile(user_id) # This will raise NotFoundException if not found
```

Ruff (0.11.9)

Local variable `profile` is assigned to but never used

Remove assignment to unused variable `profile`

(F841)